tests/test_report.py

```diff
@@ -0,0 +1,108 @@
+from src.pcost import portfolio_cost
+import pytest
+from src.report import (
+    print_report,
+    read_portfolio,
+    read_prices,
+    calc_gain_loss,
+    make_report,
+)
+
+
+def test_read_portfolio():
+    """
+    Tests if read_portfolio constructs each entry in the list correctly.
+    """
+    portfolio = read_portfolio("data/portfolio.csv")
+    expected = [
+        {"name": "AA", "shares": 100, "price": 32.2},
+        {"name": "IBM", "shares": 50, "price": 91.1},
+        {"name": "CAT", "shares": 150, "price": 83.44},
+        {"name": "MSFT", "shares": 200, "price": 51.23},
+        {"name": "GE", "shares": 95, "price": 40.37},
+        {"name": "MSFT", "shares": 50, "price": 65.1},
+        {"name": "IBM", "shares": 100, "price": 70.44},
+    ]
+
+    assert len(portfolio) == len(expected)
+    assert all([p == e for p, e in zip(portfolio, expected)])
```

Comment:
Instead of those 2 asserts you can simply do:
```python
assert portfolio == expected
```

2 lists are considered as equals if:
1. same length (similar to the first assert)
2. item of list 1 at index i is equal to item of list 2 for index i for i = 0 to end (similar to second check)